src/models/Scheduler.php

```diff
@@ -518,6 +510,27 @@ private function isFutureOrExplode(DateTime|DateTimeImmutable|false $date): void
         }
     }
 
+    /**
+     * Check that the item has been created in the last minutes (GRACE_PERIOD_MINUTES)
+     * Users can't delete events in the past (see #5596) unless in this span of time.
+     */
+    private function isInGracePeriod(DateTimeImmutable $createdAt): bool
+    {
+        if ($this->Items->Users->isAdmin) {
```

Comment:
this check must not appear here. this function is named `isInGracePeriod`. It must tell you if you are in grace period yes or no. The fact that the user is Admin has nothing to do here.